app/src/main/java/org/kiwix/kiwixmobile/wifi_hotspot/WifiHotspotManager.java

```diff
@@ -0,0 +1,88 @@
+package org.kiwix.kiwixmobile.wifi_hotspot;
+
+import android.content.Context;
+import android.net.wifi.WifiConfiguration;
+import android.net.wifi.WifiManager;
+import android.os.Build;
+import android.os.Handler;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+/**
+ * WifiHotstopManager class makes use of the Android's WifiManager and WifiConfiguration class
+ * to implement the wifi hotspot feature.
+ * Created by Adeel Zafar on 28/5/2019.
+ */
+
+public class WifiHotspotManager {
+  private final WifiManager wifiManager;
+  private WifiManager.LocalOnlyHotspotReservation hotspotReservation;
+  private HotspotStateListener hotspotStateListener;
+  private static final String TAG = "WifiHotspotManager";
+
+  public WifiHotspotManager(@NonNull Context context) {
+    wifiManager =
```

Comment:
you should just inject the wifiManager and not worry about context at all

Reply:
Fixed. a0ed6c3cd05c316a041c43fa7dfe4af97b32b5cb

Reply:
We still need `context` for
`    wifiManager =
        (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE); `

Reply:
Check if that's correct 9e5ef5fba4d9dbcf3789c75261d6af1bc022f2fd

Reply:
I left some comments, I'll resolve this one and shift attention to those